From ae4a39474df08d56064453f0fb6c2272e6c3dc8b Mon Sep 17 00:00:00 2001
From: Mark Reynolds <mreynolds@redhat.com>
Date: Fri, 6 Feb 2026 15:13:30 -0500
Subject: [PATCH] Issue 7221 - CI tests - fix some flaky tests

Description:

Try to harden some of the flaky tests with sleeps and more relaxed contraints

Relates: https://github.com/389ds/389-ds-base/issues/7221

Reviewed by: spichugi(Thanks!)
---
 dirsrvtests/tests/suites/acl/acivattr_test.py              | 7 +++++--
 dirsrvtests/tests/suites/import/regression_test.py         | 2 +-
 .../suites/replication/wait_for_async_feature_test.py      | 2 +-
 dirsrvtests/tests/suites/retrocl/basic_test.py             | 2 +-
 4 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/dirsrvtests/tests/suites/acl/acivattr_test.py b/dirsrvtests/tests/suites/acl/acivattr_test.py
index d55eea023..efe2d5ef1 100644
--- a/dirsrvtests/tests/suites/acl/acivattr_test.py
+++ b/dirsrvtests/tests/suites/acl/acivattr_test.py
@@ -1,12 +1,12 @@
 # --- BEGIN COPYRIGHT BLOCK ---
-# Copyright (C) 2019 Red Hat, Inc.
+# Copyright (C) 2026 Red Hat, Inc.
 # All rights reserved.
 #
 # License: GPL (version 3 or any later version).
 # See LICENSE for details.
 # --- END COPYRIGHT BLOCK ---
 
-import pytest, os, ldap
+import pytest, os, ldap, time
 from lib389._constants import DEFAULT_SUFFIX, PW_DM
 from lib389.idm.user import UserAccount
 from lib389.idm.organization import Organization
@@ -190,6 +190,8 @@ def test_positive(topo, _add_user, aci_of_user, user, entry, aci):
     """
     # set aci
     Domain(topo.standalone, DNBASE).set("aci", aci)
+    time.sleep(.5)
+
     # create connection
     conn = UserAccount(topo.standalone, user).bind(PW_DM)
     # according to the aci , user will  be able to change description
@@ -242,6 +244,7 @@ def test_negative(topo, _add_user, aci_of_user, user, entry, aci):
     """
     # set aci
     Domain(topo.standalone, DNBASE).set("aci", aci)
+    time.sleep(.5)
     # create connection
     conn = UserAccount(topo.standalone, user).bind(PW_DM)
     # according to the aci , user will not be able to change description
diff --git a/dirsrvtests/tests/suites/import/regression_test.py b/dirsrvtests/tests/suites/import/regression_test.py
index 0e3ba1930..25a7f359d 100644
--- a/dirsrvtests/tests/suites/import/regression_test.py
+++ b/dirsrvtests/tests/suites/import/regression_test.py
@@ -687,7 +687,7 @@ def test_ldif2db_after_backend_create(topo, verify):
     import_time_2 = create_backend_and_import(instance, ldif_file_2, 'o=test_2', 'test_2')
 
     log.info('Import times should be approximately the same')
-    assert abs(import_time_1 - import_time_2) < 15
+    assert abs(import_time_1 - import_time_2) < 20
 
 
 def test_ldif_missing_suffix_entry(topo, request, verify):
diff --git a/dirsrvtests/tests/suites/replication/wait_for_async_feature_test.py b/dirsrvtests/tests/suites/replication/wait_for_async_feature_test.py
index c5ab585e4..84ce1ca2b 100644
--- a/dirsrvtests/tests/suites/replication/wait_for_async_feature_test.py
+++ b/dirsrvtests/tests/suites/replication/wait_for_async_feature_test.py
@@ -26,7 +26,7 @@ log = logging.getLogger(__name__)
 installation1_prefix = None
 
 # Expected minimum and maximum number of async result in usual cases
-USUAL_MIN_AP = 3
+USUAL_MIN_AP = 2
 USUAL_MAX_AP = 11
 
 @pytest.fixture(params=[(None, (USUAL_MIN_AP, USUAL_MAX_AP)),
diff --git a/dirsrvtests/tests/suites/retrocl/basic_test.py b/dirsrvtests/tests/suites/retrocl/basic_test.py
index b53a60851..2fce72049 100644
--- a/dirsrvtests/tests/suites/retrocl/basic_test.py
+++ b/dirsrvtests/tests/suites/retrocl/basic_test.py
@@ -492,7 +492,7 @@ def test_retrocl_trimming_entries(topology_st):
         if inst.searchErrorsLog("trim_changelog: removed "):
             log.info(f'Trimming detected after {attempt * 6} seconds')
             break
-    
+
     log.info('Verify trimming occurred by checking error log')
     assert inst.searchErrorsLog("trim_changelog: removed ")
 
-- 
2.52.0

